### PR TITLE
[16.0][ADD] base_tier_validation_definition_server_action

### PR DESCRIPTION
**Pull request by JasminSForgeFlow**
Add a new type server action for tier validation definition

@ForgeFlow

### CODE REVIEW

**Review by PauBForgeFlow — APPROVED** [review]
Code and functional review OK, LGTM!

File: base_tier_validation_definition_server_action/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+To define the server action,
+* By python code choose the **Server Action** option in the Definition field.
```
Comment (LoisRForgeFlow) [inline base_tier_validation_definition_server_action/readme/USAGE.rst]: Could you elaborate a bit more how to configure?

- What content/code is expected in the server action? (to return a variable or to return a boolean?)
- Which model should you use to evaluate the server action? the same as the tier.definition? is possible to use a different one?

Comment (JasminSForgeFlow) [reply base_tier_validation_definition_server_action/readme/USAGE.rst]: @LoisRForgeFlow its updated.

Thanks.

File: base_tier_validation_definition_server_action/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+On each Tier Definition
+
+* Choose the **Server Action** option in the Definition.
+    * Server action should return True in order to add validation on object
```
Comment (LoisRForgeFlow) [inline base_tier_validation_definition_server_action/readme/CONFIGURE.rst]: This is not correct looking at [tests](https://github.com/OCA/server-ux/pull/783/files#diff-a1a1aa2035bae8b54428317a1b58cc81bc0344b5ce732a0695bf208fae6c26d8R74) and my functional review. The `return` keyword is actaully forbidden in server actions. It looks like an `action` variable is expected, isn't it?

Comment (JasminSForgeFlow) [reply base_tier_validation_definition_server_action/readme/CONFIGURE.rst]: Yes, it is

Comment (JasminSForgeFlow) [reply base_tier_validation_definition_server_action/readme/CONFIGURE.rst]: Its updated,

Thanks

**Review by LoisRForgeFlow — APPROVED** [review]
All good now, code and functional review :+1: 

/ocabot merge nobump

**Pull request merged**